### PR TITLE
Updating README and Samples for Event Grid

### PR DESCRIPTION
**Pull request by kerri-lee**

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/README.md [hunk 1]
```diff
@@ -1,90 +1,251 @@
...
+```csharp
+string sasToken = EventGridPublisherClient.BuildSharedAccessSignature(
+    "<endpoint>",
+    DateTimeOffset.UtcNow.AddMinutes(60),
+    new AzureKeyCredential("<access-key>"));
+
+EventGridPublisherClient client = new EventGridPublisherClient(
+    "<endpoint>",
+    new EventGridSharedAccessSignatureCredential(sasToken));
+```
```
Comment (kerri-lee) [inline sdk/eventgrid/Azure.Messaging.EventGrid/README.md:81]: Remove duplicate

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventgrid/Azure.Messaging.EventGrid/README.md [hunk 2]
```diff
@@ -207,24 +213,36 @@ foreach (CloudEvent cloudEvent in cloudEvents)
...
+### Service Responses
 `SendEvents()` returns a HTTP response code from the service. A `RequestFailedException` is thrown as a service response for any unsuccessful requests. The exception contains information about what response code was returned from the service.
 
-#### Deserializing Event Data
+### Deserializing Event Data
 - An `InvalidCastException` will be thrown during `GetData<T>()` if the event data cannot be cast to the specified type.
 
 - An `InvalidOperationException` will be thrown during `GetData<T>()` if a custom serializer is passed into `GetData<T>()` with non-serialized event data (for example, if the event was created by the user and not created by parsing from JSON).
 
+### Setting up console logging
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/README.md:235]: We should link out to the documentation for this I think. Look at ServiceBus for example.

Comment (kerri-lee) [reply sdk/eventgrid/Azure.Messaging.EventGrid/README.md:235]: For the exceptions, or the logging?

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/README.md:235]: The console logging. We link out to central documentation for this in other libraries.

Comment (kerri-lee) [reply sdk/eventgrid/Azure.Messaging.EventGrid/README.md:235]: Oh the link is at the bottom (line 244)

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/README.md:235]: Here is the format in SB - https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/servicebus/Azure.Messaging.ServiceBus#other-exceptions

**Pull request merged**